aws/resource_aws_route53_record.go

```diff
@@ -459,30 +459,21 @@ func resourceAwsRoute53RecordCreate(d *schema.ResourceData, meta interface{}) er
 }
 
 func changeRoute53RecordSet(conn *route53.Route53, input *route53.ChangeResourceRecordSetsInput) (interface{}, error) {
-	wait := resource.StateChangeConf{
-		Pending:    []string{"rejected"},
-		Target:     []string{"accepted"},
-		Timeout:    5 * time.Minute,
-		MinTimeout: 1 * time.Second,
-		Refresh: func() (interface{}, string, error) {
-			resp, err := conn.ChangeResourceRecordSets(input)
-			if err != nil {
-				if r53err, ok := err.(awserr.Error); ok {
-					if r53err.Code() == "PriorRequestNotComplete" {
-						// There is some pending operation, so just retry
-						// in a bit.
-						return nil, "rejected", nil
-					}
-				}
-
-				return nil, "failure", err
-			}
-
-			return resp, "accepted", nil
-		},
+	var out *route53.ChangeResourceRecordSetsOutput
+	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+		var err error
+		out, err = conn.ChangeResourceRecordSets(input)
+		if isAWSErr(err, "NoSuchHostedZone", "") {
+			log.Print("[DEBUG] Hosted Zone not found, retrying...")
+			return resource.RetryableError(err)
+		}
+		return resource.NonRetryableError(err)
```

Comment:
Nit: In the future we will likely require the usage of `resource.NonRetryableError()`/`resource.RetryableError()` with non-nil errors: 

```suggestion
		if err != nil {
			return resource.NonRetryableError(err)
		}

		return nil
```

See also: https://github.com/hashicorp/terraform-plugin-sdk/pull/199

Reply:
+1
fixed with https://github.com/terraform-providers/terraform-provider-aws/pull/11895/commits/f5cf401f5cec685033d519c03233bdce3f112173